src/test/java/org/opentripplanner/routing/fares/OrcaFareServiceTest.java

```diff
@@ -0,0 +1,162 @@
+     * charged a cash rate. At this point, the two hour window will start again so the final transfer will be
+     * charged at a discount rate... if using Orca.
+     */
+    @Test
+    public void calculateFareThatExceedsTwoHourFreeTransferWindow() {
+        List<Ride> rides = Lists.newArrayList();
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 0));
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 30));
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 60));
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 90));
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 120));
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 150));
+
+        calculateFare(rides, Fare.FareType.regular, DEFAULT_RIDE_PRICE_IN_CENTS * 6);
+        calculateFare(rides, Fare.FareType.senior, DEFAULT_RIDE_PRICE_IN_CENTS * 6);
+        calculateFare(rides, Fare.FareType.youth, DEFAULT_RIDE_PRICE_IN_CENTS * 6);
+        calculateFare(rides, Fare.FareType.electronicSpecial, 100f + 0f + 0f + 0f + DEFAULT_RIDE_PRICE_IN_CENTS + 100f);
+        calculateFare(rides, Fare.FareType.electronicRegular, DEFAULT_RIDE_PRICE_IN_CENTS + 0f + 0f + 0f +
+            DEFAULT_RIDE_PRICE_IN_CENTS + DEFAULT_RIDE_PRICE_IN_CENTS);
+        calculateFare(rides, Fare.FareType.electronicSenior, 100f + 0f + 0f + 0f + DEFAULT_RIDE_PRICE_IN_CENTS + 100f);
+        calculateFare(rides, Fare.FareType.electronicYouth, 200f + 0f + 0f + 0f + DEFAULT_RIDE_PRICE_IN_CENTS + 200f);
+    }
+
+    /**
+     * This trip starts with a cash fare so the free transfer window doesn't start until the second transfer. Therefore,
+     * all subsequent transfers will come under one transfer window and only one Orca discount charge will apply.
+     */
+    @Test
+    public void calculateFareThatStartsWithACashFare() {
+        List<Ride> rides = Lists.newArrayList();
+        rides.add(getRide(WASHINGTON_STATE_FERRIES_AGENCY_ID, 0));
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 30));
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 60));
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 90));
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 120));
+        rides.add(getRide(KITSAP_TRANSIT_AGENCY_ID, 149));
+
+        calculateFare(rides, Fare.FareType.regular, DEFAULT_RIDE_PRICE_IN_CENTS * 6);
+        calculateFare(rides, Fare.FareType.senior, DEFAULT_RIDE_PRICE_IN_CENTS * 6);
+        calculateFare(rides, Fare.FareType.youth, DEFAULT_RIDE_PRICE_IN_CENTS * 6);
+        calculateFare(rides, Fare.FareType.electronicSpecial, DEFAULT_RIDE_PRICE_IN_CENTS + 100f + 0f + 0f + 0f + 0f);
+        calculateFare(rides, Fare.FareType.electronicRegular, DEFAULT_RIDE_PRICE_IN_CENTS +
+            DEFAULT_RIDE_PRICE_IN_CENTS + 0f + 0f + 0f);
+        calculateFare(rides, Fare.FareType.electronicSenior, DEFAULT_RIDE_PRICE_IN_CENTS + 100f + 0f + 0f + 0f + 0f);
+        calculateFare(rides, Fare.FareType.electronicYouth, DEFAULT_RIDE_PRICE_IN_CENTS + 200f + 0f + 0f + 0f + 0f);
```

Comment:
I'm pretty sure the youth reduced fare for Kitsap is $1. See [here](https://www.kitsaptransit.com/fares/fares).